sources/src/resources/init-scripts/gradle-actions.inject-develocity.init.gradle

```diff
@@ -121,21 +123,35 @@ if (GradleVersion.current() < GradleVersion.version('6.0')) {
                     logger.lifecycle("Applying $BUILD_SCAN_PLUGIN_CLASS via init script")
                     applyPluginExternally(pluginManager, BUILD_SCAN_PLUGIN_CLASS)
                     if (geUrl) {
-                        logger.lifecycle("Connection to Develocity: $geUrl, allowUntrustedServer: $geAllowUntrustedServer")
+                        logger.lifecycle("Connection to Develocity: $geUrl, allowUntrustedServer: $geAllowUntrustedServer, captureTaskInputFiles: $geCaptureTaskInputFiles")
                         buildScan.server = geUrl
                         buildScan.allowUntrustedServer = geAllowUntrustedServer
                     }
                     buildScan.publishAlways()
+                    if (isAtLeast(gePluginVersion, '2.1') && atLeastGradle5) {
+                        if (isAtLeast(gePluginVersion, '3.7')) {
+                            buildScan.capture.taskInputFiles = geCaptureTaskInputFiles
+                        } else {
+                            buildScan.captureTaskInputFiles = geCaptureTaskInputFiles
+                        }
+                    }
                     if (buildScan.metaClass.respondsTo(buildScan, 'setUploadInBackground', Boolean)) buildScan.uploadInBackground = buildScanUploadInBackground  // uploadInBackground not available for build-scan-plugin 1.16
                     buildScan.value CI_AUTO_INJECTION_CUSTOM_VALUE_NAME, CI_AUTO_INJECTION_CUSTOM_VALUE_VALUE
                 }
 
                 if (geUrl && geEnforceUrl) {
                     pluginManager.withPlugin(BUILD_SCAN_PLUGIN_ID) {
                         afterEvaluate {
-                            logger.lifecycle("Enforcing Develocity: $geUrl, allowUntrustedServer: $geAllowUntrustedServer")
+                            logger.lifecycle("Enforcing Develocity: $geUrl, allowUntrustedServer: $geAllowUntrustedServer, captureTaskInputFiles: $geCaptureTaskInputFiles")
                             buildScan.server = geUrl
                             buildScan.allowUntrustedServer = geAllowUntrustedServer
+                            if (isAtLeast(gePluginVersion, '2.1') && atLeastGradle5) {
+                                if (isAtLeast(gePluginVersion, '3.7')) {
+                                    buildScan.capture.taskInputFiles = geCaptureTaskInputFiles
+                                } else {
+                                    buildScan.captureTaskInputFiles = geCaptureTaskInputFiles
+                                }
+                            }
```

Comment:
I think this block should be moved outside of the `afterEvaluate`. Setting aside the fact that the agent does not let you configure the fingerprint flag that late, it would not be correct to do it here anyway. Setting `DEVELOCITY_ENFORCE_URL` should only override `server` and `allowUntrustedServer`.

Reply:
Yup, seems to have fixed the 5.6.4 issue